src/NHibernate/Loader/JoinWalker.cs

```diff
@@ -840,6 +851,21 @@ protected JoinFragment MergeOuterJoins(IList<OuterJoinableAssociation> associati
 			return outerjoin;
 		}
 
+		private IList<OuterJoinableAssociation> GetSortedAssociations(IList<OuterJoinableAssociation> associations)
+		{
+			var indexes = GetTopologicalSortOrder(associations);
+			if (indexes == null)
```

Comment:
Can we remove null-driven logic here? Eg move conditions and dependents calculation from `GetTopologicalSortOrder` to here.

Reply:
Done